CoreScriptsRoot/Modules/Settings/SettingsHub.lua

```diff
@@ -272,7 +301,61 @@ local function CreateSettingsHub()
 			Parent = this.Shield
 		};
 
-		local barHeight = 60
+		--Bumper hints
+		this.BumperHintBar = utility:Create("Frame") {
+			Parent = this.Shield,
+			Visible = false,
+			Name = "BumperHints",
+			BackgroundTransparency = 1,
+
+			ZIndex = SETTINGS_BASE_ZINDEX + 1,
+
+			Position = UDim2.new(0, 0, 0, 0),
+			Size = UDim2.new(1, 0, 0, 64)
+		}
+	
+		this.BumperHintLeft = utility:Create("ImageLabel") {
+			Parent = this.BumperHintBar,
+			Visible = true,
+			Name = "BumperHintLeft",
+			Image = XBOX_LIGHT_SPRITESHEET,
+			ImageRectOffset = BUMPER_LEFT_SPRITE_OFFSET,
+			ImageRectSize = Vector2.new(BUMPER_IMG_WIDTH, BUMPER_IMG_HEIGHT),
+			BackgroundTransparency = 1,
+
+			ZIndex = SETTINGS_BASE_ZINDEX + 1,
+			Position = UDim2.new(0, 0, 0, 0),
+			Size = UDim2.new(0, BUMPER_IMG_WIDTH, 0, BUMPER_IMG_HEIGHT)
+		}
+		this.BumperHintRight = utility:Create("ImageLabel") {
+			Parent = this.BumperHintBar,
+			Visible = true,
+			Name = "BumperHintRight",
+			Image = XBOX_LIGHT_SPRITESHEET,
+			ImageRectOffset = BUMPER_RIGHT_SPRITE_OFFSET,
+			ImageRectSize = Vector2.new(BUMPER_IMG_WIDTH, BUMPER_IMG_HEIGHT),
+			BackgroundTransparency = 1,
+
+			ZIndex = SETTINGS_BASE_ZINDEX + 1,
+			Position = UDim2.new(1, -BUMPER_IMG_WIDTH, 0, 0),
+			Size = UDim2.new(0, BUMPER_IMG_WIDTH, 0, BUMPER_IMG_HEIGHT)
+		}
+
+		local gamepadInputTypes = {
+			[Enum.UserInputType.Gamepad1] = true,
+			[Enum.UserInputType.Gamepad2] = true,
+			[Enum.UserInputType.Gamepad3] = true,
+			[Enum.UserInputType.Gamepad4] = true,
+		}
+		UserInputService.InputBegan:connect(function(inputObj)
+			if gamepadInputTypes[inputObj.UserInputType] and this.BumperHintBar.Size.Y.Offset > MINIMUM_BUMPER_HINT_HEIGHT then
```

Comment:
What is the scenario where we shrink the bumper bar so much we don't want to show the hints anymore?
Can we also call this function when we resize the BumperHintBar?